test/hotspot/jtreg/compiler/loopopts/superword/ReductionPerf.java

```diff
@@ -21,242 +21,568 @@
+import jdk.test.lib.Utils;
 
 public class ReductionPerf {
-    public static void main(String[] args) throws Exception {
-        int[] a1 = new int[8 * 1024];
-        int[] a2 = new int[8 * 1024];
-        int[] a3 = new int[8 * 1024];
-        long[] b1 = new long[8 * 1024];
-        long[] b2 = new long[8 * 1024];
-        long[] b3 = new long[8 * 1024];
-        float[] c1 = new float[8 * 1024];
-        float[] c2 = new float[8 * 1024];
-        float[] c3 = new float[8 * 1024];
-        double[] d1 = new double[8 * 1024];
-        double[] d2 = new double[8 * 1024];
-        double[] d3 = new double[8 * 1024];
-
-        ReductionInit(a1, a2, a3, b1, b2, b3, c1, c2, c3, d1, d2, d3);
-
-        int sumIv = sumInt(a1, a2, a3);
-        long sumLv = sumLong(b1, b2, b3);
-        float sumFv = sumFloat(c1, c2, c3);
-        double sumDv = sumDouble(d1, d2, d3);
-        int mulIv = prodInt(a1, a2, a3);
-        long mulLv = prodLong(b1, b2, b3);
-        float mulFv = prodFloat(c1, c2, c3);
-        double mulDv = prodDouble(d1, d2, d3);
-
-        int sumI = 0;
-        long sumL = 0;
-        float sumF = 0.f;
-        double sumD = 0.;
-        int mulI = 0;
-        long mulL = 0;
-        float mulF = 0.f;
-        double mulD = 0.;
-
-        System.out.println("Warmup ...");
-        long start = System.currentTimeMillis();
-
-        for (int j = 0; j < 2000; j++) {
-            sumI = sumInt(a1, a2, a3);
-            sumL = sumLong(b1, b2, b3);
-            sumF = sumFloat(c1, c2, c3);
-            sumD = sumDouble(d1, d2, d3);
-            mulI = prodInt(a1, a2, a3);
-            mulL = prodLong(b1, b2, b3);
-            mulF = prodFloat(c1, c2, c3);
-            mulD = prodDouble(d1, d2, d3);
-        }
-
-        long stop = System.currentTimeMillis();
-        System.out.println(" Warmup is done in " + (stop - start) + " msec");
-
-        if (sumIv != sumI) {
-            System.out.println("sum int:    " + sumIv + " != " + sumI);
-        }
-        if (sumLv != sumL) {
-            System.out.println("sum long:   " + sumLv + " != " + sumL);
-        }
-        if (sumFv != sumF) {
-            System.out.println("sum float:  " + sumFv + " != " + sumF);
-        }
-        if (sumDv != sumD) {
-            System.out.println("sum double: " + sumDv + " != " + sumD);
-        }
-        if (mulIv != mulI) {
-            System.out.println("prod int:    " + mulIv + " != " + mulI);
-        }
-        if (mulLv != mulL) {
-            System.out.println("prod long:   " + mulLv + " != " + mulL);
-        }
-        if (mulFv != mulF) {
-            System.out.println("prod float:  " + mulFv + " != " + mulF);
-        }
-        if (mulDv != mulD) {
-            System.out.println("prod double: " + mulDv + " != " + mulD);
+    static final int RANGE = 8192;
+    static Random rand = Utils.getRandomInstance();
+
+    public static void main(String args[]) {
+        int iter_warmup = 2_000;
+        int iter_perf   = 5_000;
```

Comment:
Did you measure total execution time of this test? 
Warmup 2_000 iterations is too big number I think. You have 8192 iterations already in tested methods. 10 should be enough to trigger compilation. May be add `-Xbatch` if you want to make sure C2 does compile it.
If reduction is not supported (MulReduction on RISC-V and ARCH64) the test will be slow.

Reply:
I reduced the iteration count to `100` and `1000`. For performance measurement they can be increased. I also added `-Xbatch`. On my laptop, the test now definately runs in less than 2 seconds, with or without SuperWord. So even platforms that do not support a feature, or SuperWord as a whole could run it decently fast.